File: 1004.最大连续1的个数III/solution.go
func longestOnes(A []int, K int) int {
	sz := len(A)
	l, r := 0, 0
	ans, cnt := 0, 0
	for l < sz {
		for r < sz && cnt+(A[r]^1) <= K {
			cnt += 1 - A[r]
			r++
		}
		ans = max(ans, r-l)
		cnt -= 1 - A[l]
		l++
	}
	return ans
}

func max(a, b int) int {
	if a >= b {
		return a
	}
	return b
}

File: 1004.最大连续1的个数III/solution.cpp
class Solution {
public:
    int longestOnes(vector<int>& A, int K) {
        int sz=A.size();
        int ans=0;
        int cnt=0;
        int l=0,r=0;
        for (l=0;l<sz;l++) {
            while (r<sz && cnt+(A.at(r)^1)<=K) {
                cnt+= A.at(r)==0 ? 1 : 0;
                r++;
            }
            ans=max(ans,r-l);
            cnt-= A.at(l)==0 ? 1 : 0;
        }
        return ans;
    }
};
